akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -151,6 +155,8 @@ private[akka] object FanIn {
 
     def isCancelled(input: Int): Boolean = cancelled(input)
 
+    def isAllCompleted(): Boolean = (0 to inputCount - 1).map(completed).forall(_ == true)
```

Comment:
Can add counter today.
> We need the test cases though anyway.

Have one test case with Source.empty.concat(Source.empty). Are you thinking about other scenarios?